### DISCUSSION TITLE
Add coroutines ,channel,just like go? or add futures just like promises async await?

### DISCUSSION THREAD

**Discussion by zhuxiujia** [vlang/v#11647] [Ideas]
* coroutines(go)  or  promises(js)  futures(rust)

Can provide hundreds of thousands of concurrent services on the server side
* channel  

Sending data Between threads or coroutines

**Comment by cristian-ilies-vasile** [comment]
@zhuxiujia 
FYI
[Proper support for distributed computing, parallelism and concurrency](https://github.com/vlang/v/issues/1868)

**Comment by ylluminate** [comment]
Discussion thread: https://discordapp.com/channels/592103645835821068/592106336838352923/679739670321692891

**Comment by dumblob** [comment]
FYI an overview: https://github.com/vlang/v/issues/3814 .
